mypy/typeanal.py

```diff
@@ -1077,6 +1194,15 @@ def tuple_type(self, items: List[Type]) -> TupleType:
         any_type = AnyType(TypeOfAny.special_form)
         return TupleType(items, fallback=self.named_type('builtins.tuple', [any_type]))
 
+    @contextmanager
+    def set_allow_param_spec_literals(self, to: bool) -> Iterator[None]:
```

Comment:
I don't generally think of a `set_` function as being a context manager, maybe a different name like `scoped_allow_param_spec_literals`?

Another option is to create a new visitor anytime we need to do something like this, but this may be annoying wihtout a dataclasses-like mutate function.